scio-examples/src/main/scala/com/spotify/scio/examples/extra/WriteDynamicExample.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.examples.extra
+
+import com.spotify.scio.ContextAndArgs
+import com.spotify.scio.examples.common.ExampleData
+import org.apache.beam.sdk.coders.StringUtf8Coder
+import org.apache.beam.sdk.io.{FileIO, TextIO}
+import org.apache.beam.sdk.transforms.Contextful
+
+// `sbt runMain "com.spotify.scio.examples.extra.WriteDynamicExample
+// --project=[PROJECT] --runner=DataflowRunner --zone=[ZONE]
+// --input=gs://apache-beam-samples/shakespeare/kinglear.txt
+// --output=[OUTPUT]"
+object WriteDynamicExample {
+  case class LinesPerCharacter(name: String, lines: Long)
+
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+
+    // Group records according to first letter of the character's name
+    // All characters starting with letter A will share an output path, etc
+    val dynamicOutput: FileIO.Write[String, LinesPerCharacter] = FileIO
+      .writeDynamic[String, LinesPerCharacter]()
+      .by(Contextful.fn[LinesPerCharacter, String]((linesPerCharacter: LinesPerCharacter) =>
```

Comment:
Guess we need the explicit lambda arg type here for scala 2.11 compatibility?

Reply:
yup